pkg/translator/translator_test.go

```diff
@@ -0,0 +1,85 @@
+package translator
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/honeycombio/hpsf/pkg/config"
+	"github.com/honeycombio/hpsf/pkg/hpsf"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	yamlv3 "gopkg.in/yaml.v3"
+)
+
+func TestGenerateConfig(t *testing.T) {
+	const inputData = `components:
+  - name: RefineryGRPC_2
+    kind: RefineryGRPC
+    ports:
+      - name: TraceOut
+        direction: output
+        type: Honeycomb
+    properties:
+      - name: Port
+        value: 4317
+        type: number
+  - name: otlp_in
+    kind: OTelReceiver
+    properties:
+      - name: GRPCPort
+        value: 9922
+      - name: HTTPPort
+        value: 1234
+  - name: otlp_out
+    kind: OTelGRPCExporter
+    properties:
+      - name: Host
+        value: myhost.com
+      - name: Port
+        value: 1234
+connections:
+  - source:
+      component: otlp_in
+      port: Traces
+      type: OTelTraces
+    destination:
+      component: otlp_out
+      port: Traces
+      type: OTelTraces`
+
+	const expectedConfig = `receivers:
```

Comment:
could the expected config live in `testdata/` file 😬

Reply:
Absosmurfly. Are you picturing a testdata/ folder at the root of the repo?

Reply:
> The simplest solution is to keep all the test data in the same directory as the code itself under a special folder called testdata. This is a special folder ignored by the compiler, but still accessible for reading during tests. 

**https://hackandsla.sh/posts/2020-11-23-golang-test-fixtures/#keeping-test-data-under-testdata**

Reply:
Moved embedded HEREDOC(ish) strings to `./testdata/` files.